examples/java/core/protocol_versions/src/main/java/org/bondlib/ProtocolVersions.java

```diff
@@ -57,6 +59,16 @@ public static void main(final String[] args) throws IOException {
             assert obj.equals(obj2) : "Roundtrip CBv2 failed";
         }
 
-        assert false : "Add marshaling example here";
+        {
+            // Here, we Marshal to CompactBinary v2.
+            final ByteArrayOutputStream output = new ByteArrayOutputStream();
+            final CompactBinaryWriter writer = new CompactBinaryWriter(output, 2);
+            Marshal.marshal(obj, writer);
+
+            final ByteArrayInputStream input = new ByteArrayInputStream(output.toByteArray());
+            final Struct obj2 = Unmarshal.unmarshal(input, Struct.BOND_TYPE).deserialize();
```

Comment:
The C# example has a useful comment above the call to unmarshal. Can we add a similar one?

                // The protocol and version are determined from the payload
                // itself.

Reply:
Done.